cli/azd/internal/useragent.go

```diff
@@ -38,6 +39,23 @@ func (userAgent *UserAgent) String() string {
 	return sb.String()
 }
 
+const (
+	templateContextKey contextKey = "template"
+)
+
+func WithTemplate(ctx context.Context, template string) context.Context {
+	return context.WithValue(ctx, templateContextKey, template)
+}
+
+func GetTemplate(ctx context.Context) string {
+	template, ok := ctx.Value(templateContextKey).(string)
+	if !ok {
+		return ""
```

Comment:
Template is purely optional to help us understand what sample templates our customers are actually using.  When people devify their projects they will likely strip out the `metadata` section of the azure.yaml or set their own value.

Either way - an empty string return value can be checked by the caller as easily as they can an empty string.  Our useragent generation already accounts for this today.  I don't feel there is any use case where we would want to throw an error because the template value has not been set.